src/transformers/trainer.py

```diff
@@ -5060,6 +5060,30 @@ def create_accelerator_and_postprocess(self):
                 "`auto_find_batch_size` isn't supported yet with DeepSpeed Zero-3. Please consider using Zero-2, Zero-1, or FSDP"
             )
 
+    def free_memory(self, *models):
+        """
+        Will release all references to the internal objects stored and call the garbage collector.
+        Should also explicitly pass in the (potentially multiple) `model` *original* references
+        to be released.
+        Example usage:
+        ```
+        >>> from transformers import Trainer
+        >>> model = MyModel()
+        >>> trainer = Trainer(model, ...)
+        >>> trainer.train()
+        >>> model = trainer.free_memory(model)
+        ```
+        """
+        # We need to have these references so they can be set to `None`
+        (
+            *models,
+            self.optimizer,
+            self.model,
+            self.deepspeed,
+            self.model_wrapped,
+        ) = self.accelerator.free_memory(*models, self.optimizer, self.model, self.deepspeed, self.model_wrapped)
+        return models
```

Comment:
Is `models` always just a tuple of `None` values? It seems unnecessary to return that!

Reply:
It's actually quite necessary because we need to free up whatever objects a user passed in to free up the references. Yay for python :melting_face:

Reply:
Those objects should be freed when nothing else refers to them, though! So the caller could just do

```python
free_memory(models)
del models
```

rather than

```python
models = free_memory(models)
```

The second one is slightly shorter, but it's much more unclear about what's happening imo (but this is a style thing, so feel free to ignore me)